### PR TITLE
Replace Xapi.exists? with X::Exercise.exists? [#2637]

### PR DESCRIPTION
**Pull request by karloxyz**
https://github.com/exercism/exercism.io/issues/2637

### CODE REVIEW

File: x/exercise.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module X
+  module Exercise
+    def self.exists?(language, slug)
```
Comment (kytrinyx) [inline x/exercise.rb]: This is the track_id, not the language.

File: x/exercise.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+      Xapi.request('tracks', language, slug).status != 404
```
Comment (kytrinyx) [inline x/exercise.rb]: We have a light-weight endpoint: http://x.exercism.io/v3/tracks/:id/exercises/:slug/exists which has a 404 or 200 status

Comment (kytrinyx) [inline x/exercise.rb]: What would be the best way to test this? All of the existing tests stub out this method, and nothing tests that it actually talks to the correct endpoint / has the correct behavior.

File: api/v1/routes/iterations.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ class Iterations < Core
           halt 401, { error: message }.to_json
         end
 
-        if !Xapi.exists?(language, slug)
+        if !X::Exercise.exists?(language, slug)
```
Comment (AlexWheeler) [inline api/v1/routes/iterations.rb]: I like how you used unless below, what do you think about using it here as well?

File: x/exercise.rb (hunk 2 above)
Comment (tejasbubane) [inline x/exercise.rb]: @kytrinyx Can actually talking to the endpoint be avoided in this case? I think we can mock `Xapi.request` here (just checking that the method was called with proper args). We'll then need to write test for [Xapi](https://github.com/exercism/exercism.io/blob/master/x/xapi.rb) where we mock `Faraday` and trust it to make the proper request.

**Pull request closed without merging**